src/otx/core/data/dataset/instance_segmentation.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTXInstanceSegDataset."""
+
+from __future__ import annotations
+
+from typing import Callable
+
+import numpy as np
+import torch
+from datumaro import DatasetSubset, Image, Polygon
+from torchvision import tv_tensors
+
+from otx.core.data.entity.base import ImageInfo
+from otx.core.data.entity.instance_segmentation import InstanceSegBatchDataEntity, InstanceSegDataEntity
+from otx.core.utils.mask_util import polygon_to_bitmap
+
+from .base import OTXDataset, Transforms
+
+
+class OTXInstanceSegDataset(OTXDataset[InstanceSegDataEntity]):
+    """OTXDataset class for instance segmentation."""
+
+    def __init__(self, dm_subset: DatasetSubset, transforms: Transforms) -> None:
+        super().__init__(dm_subset, transforms)
+        self.use_polygon = True
+
+    def _get_item_impl(self, index: int) -> InstanceSegDataEntity | None:
+        item = self.dm_subset.get(id=self.ids[index], subset=self.dm_subset.name)
+        img = item.media_as(Image)
+        img_data, img_shape = self._get_img_data_and_shape(img)
+
+        gt_bboxes, gt_labels, gt_masks, gt_polygons = [], [], [], []
+
+        for annotation in item.annotations:
+            if isinstance(annotation, Polygon):
+                bbox = np.array(annotation.get_bbox(), dtype=np.float32)
+                gt_bboxes.append(bbox)
+                gt_labels.append(annotation.label)
+
+                if self.use_polygon:
```

Comment:
I've implemented a data module configuration specifically for Instance Segmentation, accessible https://github.com/eugene123tw/training_extensions/blob/eugene/v2-inst-seg/src/otx/core/config/data.py#L45

This addition anticipates the possibility of integrating a future instance segmentation algorithm that handles masks more effectively than the current polygon method. Users may prefer training with direct masks instead of polygons in such instances.

Providing users with these two options offers flexibility and accommodating potential improvement in instance segmentation techniques.